runners/flink/src/main/java/org/apache/beam/runners/flink/execution/SingletonSdkHarnessManager.java

```diff
@@ -76,25 +75,20 @@ private static SingletonSdkHarnessManager create() {
     this.jobResourceManagerFactory = jobResourceManagerFactory;
   }
 
-  @Nullable
-  private JobResourceManager jobResourceManager = null;
-
   @Override
   public synchronized EnvironmentSession getSession(
       ProvisionApi.ProvisionInfo jobInfo,
       RunnerApi.Environment environment,
       ArtifactSource artifactSource) throws Exception {
 
-    if (jobResourceManager == null) {
-      jobResourceManager =
-          jobResourceManagerFactory.create(
-              jobInfo,
-              environment,
-              artifactSource,
-              serverFactory,
-              executorService);
-      jobResourceManager.start();
-    }
+    JobResourceManager jobResourceManager =
```

Comment:
We'll definitely need to clean up lifecycle management down the line. Please add a TODO to fix this.

CC: @axelmagn